src/redisearch_rs/inverted_index/src/raw_doc_ids_only.rs

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use std::io::{Cursor, Seek, Write};
+
+use ffi::t_docId;
+
+use crate::{Decoder, Encoder, RSIndexResult};
+
+/// Encode and decode only the raw document ID delta without any compression.
+///
+/// The delta is encoded as a raw 4-byte value.
+/// This is different from the regular [`crate::doc_ids_only::DocIdsOnly`] encoder which uses varint encoding.
+#[derive(Default)]
+pub struct RawDocIdsOnly;
+
+impl Encoder for RawDocIdsOnly {
+    type Delta = u32;
+
+    fn encode<W: Write + Seek>(
+        &mut self,
+        mut writer: W,
+        delta: Self::Delta,
+        _record: &RSIndexResult,
+    ) -> std::io::Result<usize> {
+        writer.write_all(&delta.to_ne_bytes())?;
+        // Wrote delta as raw 4-bytes word
+        Ok(4)
+    }
+}
+
+impl Decoder for RawDocIdsOnly {
+    fn decode<'a>(
+        &self,
+        cursor: &mut Cursor<&'a [u8]>,
+        base: t_docId,
+    ) -> std::io::Result<RSIndexResult<'a>> {
+        let mut delta_bytes = [0u8; 4];
+        std::io::Read::read_exact(cursor, &mut delta_bytes)?;
```

Comment:
Is there no way to initially have `delta_bytes` as an undefined `[u8; 4]` (not initialized to 0)? Probably doesn't matter much, but since on a successful read we overwrite the entry array, will Rust optimize this?